src/main/java/sparksoniq/spark/iterator/flowr/GroupByClauseSparkIterator.java

```diff
@@ -160,6 +161,19 @@ private void setAllLocalResults() {
                                     expression.getIteratorMetadata().getExpressionMetadata()
                             );
                         }
+                        if (resultItem.isDuration())
```

Comment:
Why are we converting duration/dateTime/binary grouping keys to strings?

Reply:
We need to do this because otherwise in local evaluation, items like time("13:20:00") and time("13:20:00Z") or items like dayTimeDuration("PT0M") and yearMonthDuration("P0Y") will have different key and they would not be grouped together. Using their long value and wrapping it into a StringItem allows us to actually compare the long values, and since they are the same for the two pairs of examples mentioned above, they would be grouped together.

Reply:
I now changed FlworKey's hashCode() function to call hashCode() on its keys instead of serialize() as before. Now there is no need for casting to string in GroupByClauseSparkIterator. 
The other change is for the overriding hashCode() function of Duration.
@ghislainfourny